RevenueCatUI/CustomerCenter/Data/LoadPromotionalOfferUseCase.swift

```diff
@@ -49,8 +49,12 @@ class LoadPromotionalOfferUseCase: LoadPromotionalOfferUseCaseType {
                 return .failure(CustomerCenterError.couldNotFindSubscriptionInformation)
             }
 
-            guard let discount = subscribedProduct.discounts.first(where: {
-                $0.offerIdentifier == promoOfferDetails.iosOfferId
+            guard let discount = subscribedProduct.discounts.first(where: { discount in
+                guard let offerIdentifier = discount.offerIdentifier else {
+                    return false
+                }
+                return offerIdentifier == promoOfferDetails.iosOfferId ||
+                       offerIdentifier.hasSuffix("_\(promoOfferDetails.iosOfferId)")
```

Comment:
Should we first search for the exact match and if it's not found, then find the suffix? I'm thinking that with the current implementation we might find a version with the suffix even if we have another one that is an exact match.

Reply:
I think that's better yes. I updated it